airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/discover/DiscoverIntegrationTest.kt

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.discover
+
+import io.airbyte.cdk.command.ConfigurationSpecification
+import io.airbyte.cdk.command.FeatureFlag
+import io.airbyte.cdk.load.command.Property
+import io.airbyte.cdk.load.test.util.ConfigurationUpdater
+import io.airbyte.cdk.load.test.util.FakeConfigurationUpdater
+import io.airbyte.cdk.load.test.util.FakeDataDumper
+import io.airbyte.cdk.load.test.util.IntegrationTest
+import io.airbyte.cdk.load.test.util.NoopDestinationCleaner
+import io.airbyte.cdk.load.test.util.NoopExpectedRecordMapper
+import io.airbyte.cdk.load.test.util.destination_process.NonDockerizedDestinationFactory
+import io.airbyte.protocol.models.v0.AirbyteConnectionStatus
+import io.airbyte.protocol.models.v0.AirbyteMessage
+import java.util.regex.Pattern
+import kotlin.test.assertEquals
+import kotlinx.coroutines.runBlocking
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertAll
+
+data class DiscoverTestConfig(
+    val configContents: String,
+    val featureFlags: Set<FeatureFlag> = emptySet(),
+    val name: String? = null,
+)
+
+abstract class DiscoverIntegrationTest<T : ConfigurationSpecification>(
+    val successConfigFilenames: List<DiscoverTestConfig>,
+    val failConfigFilenamesAndFailureReasons: Map<DiscoverTestConfig, Pattern>,
```

Comment:
is discover expected to ever fail on a valid config? i.e. is there a case where `check` succeeds but `discover` fails?

(if not, probably makes sense to remove the failConfigs entirely)

Reply:
I deleted this. As I was updated the test below, it was not clear what we be a relevant assertion.